numpy/_core/src/multiarray/descriptor.c

```diff
@@ -2025,6 +2025,7 @@ arraydescr_dealloc(PyArray_Descr *self)
 {
     Py_XDECREF(self->typeobj);
     if (!PyDataType_ISLEGACY(self)) {
+        Py_TYPE(self)->tp_free((PyObject *)self);
```

Comment:
Arg, had already fixed this in my (still open) PR.  But I guess it doesn't matter, let's see if it has a collision :).